## Result Exavation

In [4]:
import pickle

with open("pickle_regression.pkl", "rb") as f:
    result = pickle.load(f)



In [5]:
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    魅力度   R-squared:                       0.595
Model:                            OLS   Adj. R-squared:                  0.567
Method:                 Least Squares   F-statistic:                     21.04
Date:                Tue, 12 Nov 2024   Prob (F-statistic):           1.53e-08
Time:                        11:15:39   Log-Likelihood:                -163.04
No. Observations:                  47   AIC:                             334.1
Df Residuals:                      43   BIC:                             341.5
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         18.6402      2.117      8.806      0.000      14.371      22.909
人口         -2.304e-06   7.99e-07     -2.883      0.006   -3.92e-06   -6.92e-07
ホテル客室数         0.0008      0.000      5.781      0.000       0.001       0.001
旅館客室数      -3.578e-05      0.000     -0.242      0.810      -0.000       0.000
==============================================================================
Omnibus:                       14.240   Durbin-Watson:                   1.048
Prob(Omnibus):                  0.001   Jarque-Bera (JB):               16.469
Skew:                           1.102   Prob(JB):                     0.000265
Kurtosis:                       4.885   Cond. No.                     6.89e+06
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 6.89e+06. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

## R-Square

In [6]:
result.rsquared

0.594830126141837

In [7]:
#Adj. R-squared
result.rsquared_adj

0.5665624605238255

## F-test

In [9]:
result.f_pvalue

1.5274367685392703e-08

有意水準を 0.05 としてモデル適合度のF検定を行った結果p値は 1.5e -08 であり、全ての偏回帰係数が0であるという帰無仮説が **＜棄却された/棄却されなかった＞**された 。すなわち、今回の分析の重回帰モデルは **＜有意に妥当な変数が含まれている/妥当な変数が含まれていない可能性がある＞** 事が分かった。

## Interpretation of partial regression coefficients

In [10]:
result.pvalues

const     3.529716e-11
人口        6.130946e-03
ホテル客室数    7.617077e-07
旅館客室数     8.098594e-01
dtype: float64

In [11]:
# pvalue is more that 0.05 ,,difficult.
#no correlation
result.params

const     18.640181
人口        -0.000002
ホテル客室数     0.000824
旅館客室数     -0.000036
dtype: float64

 有意水準を下回った変数の内、偏回帰係数の付合が負となったのは人口、…であり、偏回帰係数の付合が負となったのはでホテル客室数あった。
> 変数人口の偏回帰係数は-0.000002であり、変数人口が1単位増加すると、目的変数は魅力度　-0.000002減少する。
> 変数ホテル客室数 の偏回帰係数は 0.000824であり、変数人口が1単位増加すると、目的変数は魅力度　 0.000824増加する。

In [ ]:
#When all puppose variale is zero

